### PR TITLE
FileTypeIcon enhancements

### PR DESCRIPTION
**Pull request by GuidoZam**
| Q               | A
| --------------- | ---
| Bug fix?        | [ ]
| New feature?    | [x]
| New sample?      | [ ]
| Related issues?  | #270 

#### What's in this Pull Request?

Added some events to the FileTypeIcon control.

The new events are the following:
- onClick
- onDoubleClick
- onMouseEnter
- onMouseLeave
- onMouseOver
- onMouseUp

### CODE REVIEW

File: src/controls/fileTypeIcon/IFileTypeIcon.ts [hunk 1]
```diff
@@ -1,3 +1,5 @@
+import { FocusEventHandler, KeyboardEventHandler, MouseEventHandler, ReactEventHandler } from "react";
```
Comment (michaelmaillot) [inline src/controls/fileTypeIcon/IFileTypeIcon.ts:1]: Please remove unused imports

File: docs/documentation/docs/controls/FileTypeIcon.md [hunk 2]
```diff
@@ -42,5 +42,11 @@ The FileTypeIcon component can be configured with the following properties:
 | path | string | no | Path to the document. If this is provided, the control will use the file extension to display the corresponding icon. |
 | size | ImageSize | no | This is a property that only needs to be used when the type is set to image. It allows you to specify the image size. small (16px), normal (20px), medium (48px) and large (96px) are possible. Use the **ImageSize** enum to get the list of available images sizes. |
 | type | IconType | yes | This property specifies is you want to use the icon font or image. Use the **IconType** enum to get the list of available icon types. |
+| onClick | React.MouseEvent\<HTMLElement> | no | Event triggered when the icon is clicked. |
+| onDoubleClick | React.MouseEvent\<HTMLElement> | no | Event triggered when the icon is double clicked. |
+| onMouseEnter | React.MouseEvent\<HTMLElement> | no | Event triggered when the mouse cursor enter the icon. |
```
Comment (michaelmaillot) [inline docs/documentation/docs/controls/FileTypeIcon.md:47]:
```suggestion
| onMouseEnter | React.MouseEvent&lt;HTMLElement&gt; | no | Event triggered when the mouse cursor enters the icon (without event bubbling). |
```

File: docs/documentation/docs/controls/FileTypeIcon.md [hunk 3]
```diff
@@ -42,5 +42,11 @@ The FileTypeIcon component can be configured with the following properties:
...
+| onMouseLeave | React.MouseEvent\<HTMLElement> | no | Event triggered when the mouse cursor leave the icon. |
```
Comment (michaelmaillot) [inline docs/documentation/docs/controls/FileTypeIcon.md:48]:
```suggestion
| onMouseLeave | React.MouseEvent&lt;HTMLElement&gt; | no | Event triggered when the mouse cursor leaves the icon. |
```

File: docs/documentation/docs/controls/FileTypeIcon.md [hunk 4]
```diff
@@ -42,5 +42,11 @@ The FileTypeIcon component can be configured with the following properties:
...
+| onMouseOver | React.MouseEvent\<HTMLElement> | no | Event triggered when the mouse cursor is over the icon. |
```
Comment (michaelmaillot) [inline docs/documentation/docs/controls/FileTypeIcon.md:49]:
```suggestion
| onMouseOver | React.MouseEvent&lt;HTMLElement&gt; | no | Event triggered when the mouse cursor enters the icon (with event bubbling). |
```

File: docs/documentation/docs/controls/FileTypeIcon.md [hunk 5]
```diff
@@ -42,5 +42,11 @@ The FileTypeIcon component can be configured with the following properties:
 | path | string | no | Path to the document. If this is provided, the control will use the file extension to display the corresponding icon. |
 | size | ImageSize | no | This is a property that only needs to be used when the type is set to image. It allows you to specify the image size. small (16px), normal (20px), medium (48px) and large (96px) are possible. Use the **ImageSize** enum to get the list of available images sizes. |
 | type | IconType | yes | This property specifies is you want to use the icon font or image. Use the **IconType** enum to get the list of available icon types. |
+| onClick | React.MouseEvent\<HTMLElement> | no | Event triggered when the icon is clicked. |
```
Comment (michaelmaillot) [inline docs/documentation/docs/controls/FileTypeIcon.md:45]:
```suggestion
| onClick | React.MouseEvent&lt;HTMLElement&gt; | no | Event triggered when the icon is clicked. |
```

File: docs/documentation/docs/controls/FileTypeIcon.md [hunk 6]
```diff
@@ -42,5 +42,11 @@ The FileTypeIcon component can be configured with the following properties:
...
+| onDoubleClick | React.MouseEvent\<HTMLElement> | no | Event triggered when the icon is double clicked. |
```
Comment (michaelmaillot) [inline docs/documentation/docs/controls/FileTypeIcon.md:46]:
```suggestion
| onDoubleClick | React.MouseEvent&lt;HTMLElement&gt; | no | Event triggered when the icon is double clicked. |
```

File: docs/documentation/docs/controls/FileTypeIcon.md [hunk 7]
```diff
@@ -42,5 +42,11 @@ The FileTypeIcon component can be configured with the following properties:
...
+| onMouseEnter | React.MouseEvent\<HTMLElement> | no | Event triggered when the mouse cursor enter the icon. |
+| onMouseLeave | React.MouseEvent\<HTMLElement> | no | Event triggered when the mouse cursor leave the icon. |
+| onMouseOver | React.MouseEvent\<HTMLElement> | no | Event triggered when the mouse cursor is over the icon. |
+| onMouseUp | React.MouseEvent\<HTMLElement> | no | Event triggered when the mouse button is released after clicked on the icon. |
```
Comment (michaelmaillot) [inline docs/documentation/docs/controls/FileTypeIcon.md:50]:
```suggestion
| onMouseUp | React.MouseEvent&lt;HTMLElement&gt; | no | Event triggered when the mouse button is released after clicked on the icon. |
```

**Review by michaelmaillot — CHANGES_REQUESTED** [review]
Hi @GuidoZam,

Thanks for submitting this PR!

I have added a few changes to make.

**Comment by GuidoZam** [comment]
Hi @michaelmaillot, thanks for the reviews. I've made the changes you asked for 👍

**Comment by michaelmaillot** [comment]
Merged manually, thank you!

**Pull request merged**